### DISCUSSION TITLE
test problem：appear ：oduleNotFoundError: No module named 'dafoam.pyDASolverIncompressible'

### DISCUSSION THREAD

**Discussion by Drawthe** [mdolab/dafoam#229] [Installation]
howdy
I follow dafoam The compilation method of (HPC) compiles dafoam without any errors during installation, but there are problems such as modulenotfounderror: no module named 'dafoam. Pydasolverincompressed' during the test. Do you have any solutions? I installed version 2.2.9 of dafoam and version - v181 of openfoam. I tried to reinstall Op The two modes of enfoam, openfoam and dafoam cannot recognize this module. I only checked dafoam using PIP list, but not dafoam Pydasolverincompressible, is it my installation problem?
I really hope to get your help

**Comment by friedenhe** [comment]
Are you sure you loaded the DAFoam environment before running the test? Also, some HPCs do not allow running parallel jobs on the head node (the DAFoam tests run in parallel), you may want to check if you can run parallel OpenFOAOM native tutorials first.

**Reply by Drawthe** [reply]
Hello!
First of all, thank you very much for your reply. Here is my installation process in HPC
My current situation is as follows: the cluster environment I use is the slurm scheduling system, which passed the openfoam native test, and the test method is CD $foam_ Tutorials / incompressible / simplefoam / pitzdaily run the blockmesh and simplefoam commands. Both commands passed the test. It is also normal to use mpirun in a cluster with a smaller example. I compiled openfoam using Intel MPI. The command used for compiling is mpiicc mpiicpc mpiifort. I did not compile it completely according to the HPC compilation command. The specific modification is edit openfoam-v1812 / wmake / rules / linux64icc / C. the CC instruction of this file is changed to mpiicc
Secondly, you mentioned whether there is an environmental problem of entering dafoam. When I test the management node, I use source env_ DAFOAM. SH appears (dafoam_venv)[ sca0449@ln121 %Bscc-a5 pitzdaily] $dafoam appears_ Venv I use PIP list, which is a module that can see dafoam The following is the content of my PIP list
Package       Version Editable project location
------------- ------- ------------------------------------------------------
baseclasses   1.2.0   /public1/home/sca0449/DAFoam/repos/baseclasses-1.2.0
cgnsutilities 2.5.0   /public1/home/sca0449/DAFoam/repos/cgnsutilities-2.5.0
Cython        0.29.21
dafoam        2.2.9   /public1/home/sca0449/dafoam-2.2.9
idwarp        2.2.1   /public1/home/sca0449/DAFoam/repos/idwarp-2.2.1
mpi4py        3.1.3
multipoint    1.2.0   /public1/home/sca0449/DAFoam/repos/multipoint-1.2.0
numpy         1.21.2
numpy-stl     2.16.0
petsc4py      3.11.0
pip           21.3.1
pygeo         1.5.0   /public1/home/sca0449/DAFoam/repos/pygeo-1.5.0
pyhyp         2.2.0   /public1/home/sca0449/DAFoam/repos/pyhyp-2.2.0
pyofm         1.2.0   /public1/home/sca0449/DAFoam/repos/pyofm-1.2.1
pyoptsparse   2.3.0   /public1/home/sca0449/DAFoam/repos/pyoptsparse-2.3.0
pyspline      1.2.0   /public1/home/sca0449/DAFoam/repos/pyspline-1.2.0
python-utils  2.5.6
scipy         1.7.1
setuptools    39.0.1
six           1.16.0
sqlitedict    1.7.0
My example of testing dafoam is in the tests directory of dafoam./ Allrun uses slurm script to run in the cluster. The missing module is the script content
#!/bin/bash
#SBATCH -p amd_256   
#SBATCH -N 1
#SBATCH -n 64      
source /public1/home/sca0449/DAFoam/ENV_DAFOAM.sh
source /public1/home/sca0449/DAFoam/env.sh
./Allrun
Again, I hope you can get your help

**Reply by friedenhe** [reply]
Check if you have some *.so file in /public1/home/sca0449/dafoam-2.2.9/dafoam. If yes, run `python` and then run `import dafoam` and see if you can import the DAFoam module. If not, try to run `import idwarp` and see if you can import other modules. If all these work, I suggest you run a tutorial in serial to see if it works.

**Reply by Drawthe** [reply]
Thank you very much for your reply. I didn't find any * So file. But I have no problem importing modules in Python as you said. Can you prove that my input file is wrong, and can you provide a complete series of tutorials. Thank you very much for your reply！
![image](https://user-images.githubusercontent.com/93109645/147780553-c384f9c6-b668-4b14-b7f8-d0ebf4cfdbd4.png)

**Reply by Drawthe** [reply]
This is added：
Is there a command to detect whether to import the module
![image](https://user-images.githubusercontent.com/93109645/147780973-7c3f3136-962f-4495-9fea-f9b460ed3638.png)

**Reply by friedenhe** [reply]
Then, you need to re-run this step: https://dafoam.github.io/mydoc_installation_source_hpc.html#install-dafoam. You should see some errors. Otherwise, the *.so files should have been generated.

**Comment by ChenYingjie123** [comment]
I have a similar problem when compiling in ubuntu, is there any solution？
![1647668231(1)](https://user-images.githubusercontent.com/101343730/159108708-12204e85-773c-4c33-8213-519704faeebe.jpg)
I have source loadDAFoam.sh before testing Allrun

**Reply by Drawthe** [reply]
You this problem I met, please completely in accordance with the https://dafoam.github.io/mydoc_installation_source_hpc.html operation, not using anaconda to install the module, need to source to install python, And the PIP calling python is not Anaconda, although Anaconda is installed with the same, but I have verified that this is not possible.

**Reply by friedenhe** [reply]
@ChenYingjie123, I am not sure why this happened. Can you try to run `pip install -e  .`? Also, make sure you can import dafoam by running: python, and then run: import dafoam

**Comment by Zcaic** [comment]
I met the same problem as you , do you have solve it?
problem: No module named 'dafoam.pyDASolverIncompressible

**Reply by friedenhe** [reply]
Check if this helps: https://dafoam.github.io/mydoc_get_started_faq.html#how-to-fix-the-no-module-named-dafoampydasolver-error

**Comment by wwb1987wave** [comment]
I have the same problem. Has you sove it? If so, can you give me some suggestions? @ChenYingJie123

**Reply by friedenhe** [reply]
@wwb1987wave @Zcaic @Drawthe @ChenYingjie123 We have identified a potential problem that may cause this error. For some Ubuntu systems, the folders sharedBins and sharedLibs may be incorrectly created under the OpenFOAM folder and cause issues. We have updated our installation guide that manually creates these two folders to avoid this issue (see the new release v3.0.2). Please try the [latest installation guide](https://dafoam.github.io/mydoc_installation_source.html) (I suggest you compile everything from scratch) and let us know if this fixes the problem. Note this release also updates the pyGeo's ffd interface, check the release note from [here](https://github.com/mdolab/dafoam/releases/tag/v3.0.2)
